### PR TITLE
Convert youth follow-ups to case notes

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: app/models/grda_warehouse/youth/youth_follow_up.rb [hunk 1]
```diff
@@ -14,17 +14,43 @@ class YouthFollowUp < GrdaWarehouseBase
...
     end
 
-    scope :visible_by?, -> (user) do
+    scope :incomplete, -> do
+      where(contacted_on: nil)
+    end
+
+    scope :upcoming, -> do
+      where(contacted_on: nil).
+        where(arel_table[:required_on].between(1.weeks.from_now.to_date..Date.yesterday))
```
Comment (awisema) [inline app/models/grda_warehouse/youth/youth_follow_up.rb]: I don't think it matters, but I find this range (future..past) confusing.

File: app/models/grda_warehouse/youth/youth_follow_up.rb [hunk 2]
```diff
@@ -14,17 +14,43 @@ class YouthFollowUp < GrdaWarehouseBase
...
+    end
+
+    scope :due, -> do
```
Comment (awisema) [inline app/models/grda_warehouse/youth/youth_follow_up.rb:36]: What happens when a follow up is over due?

File: app/views/clients/youth/case_managements/_form.haml [hunk 3]
```diff
@@ -15,6 +17,11 @@
         } else {
           $('.jOtherHousingStatus').hide()
         }
+        if ($('.jHousingStatus').val() == 'This youth is currently in stable housing') {
```
Comment (awisema) [inline app/views/clients/youth/case_managements/_form.haml]: Can you use `stably_housed_string` here?

File: db/warehouse/migrate/20201006194015_convert_existing_youth_follow_ups.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        housing_status: follow_up.housing_status,
+        zip_code: follow_up.zip_code,
+      )
+    end
+
+    # Loop over follow-ups and see if there is a case management note on or after the
+    # required_on date
+    # update follow-up with case management note data
+    GrdaWarehouse::Youth::YouthFollowUp.preload(:case_managements).find_each do |follow_up|
+      next_note = follow_up.case_managements.select do |note|
```
Comment (awisema) [inline db/warehouse/migrate/20201006194015_convert_existing_youth_follow_ups.rb:32]: `next_note = follow_up.case_managements.order(engaged_on: :asc).detect { |note|  note.required_on <= note.engaged_on }`?

**Review by awisema — CHANGES_REQUESTED** [review]

File: app/models/grda_warehouse/youth/youth_follow_up.rb (hunk 1 above)
Comment (eanders) [reply app/models/grda_warehouse/youth/youth_follow_up.rb]: huh, totally missed that.

File: app/models/grda_warehouse/youth/youth_follow_up.rb (hunk 2 above)
Comment (eanders) [reply app/models/grda_warehouse/youth/youth_follow_up.rb:36]: There will be a report to highlight them, for now, they get a banner stating it is due.

File: db/warehouse/migrate/20201006194015_convert_existing_youth_follow_ups.rb (hunk 4 above)
Comment (eanders) [reply db/warehouse/migrate/20201006194015_convert_existing_youth_follow_ups.rb:32]: calling the relation with an order forces extra queries, the preload with a ruby select keeps it in RAM.  (unless you are proposing a functional change, I think we can leave it as is.

**Review by awisema — APPROVED** [review]

**Pull request merged**